main.js

```diff
@@ -0,0 +1,515 @@
+/*jshint esversion: 6 */
+const colorList = ["black", "white", "red", "blue", "green", "orange", "yellow", "purple", "brown", "crimson"];
+
+const DEFAULT_GRID_X = 100;
+const DEFAULT_GRID_Y = 50;
+
+const COLORS_PER_ROW = 10;
+
+let currentColor;
+let currentTool;
+let gridSize = [DEFAULT_GRID_X, DEFAULT_GRID_Y];
+let gridEnabled = false;
+
+function resizePalette(x, y) {
+  gridSize = [x, y];
+
+  savePaletteSize(x, y);
+  $(".palette").empty();
+  initPalette([x, y]);
+  initColors(colorList);
+  initTools();
+  initFooter();
+  initGrid();
+}
+
+function initTools() {
+  currentTool = "pencil";
+
+  let $row = $("<div>");
+  $row.addClass("tool-row");
+
+  let $pencilTool = $("<img>");
+  $pencilTool.attr("src", "images/pencil-tool.png");
+  $pencilTool.attr("width", "40px");
+  $pencilTool.attr("id", "pencil");
+  $pencilTool.addClass("selected-tool");
+  $pencilTool.addClass("tool");
+
+  let $fillTool = $("<img>");
+  $fillTool.attr("src", "images/fill-tool.png");
+  $fillTool.attr("width", "40px");
+  $fillTool.attr("id", "fill");
+  $fillTool.addClass("tool");
+
+  let $colorTool = $("<img>");
+  $colorTool.attr("src", "images/color-selector.png");
+  $colorTool.attr("width", "40px");
+  $colorTool.attr("id", "color-selector");
+  $colorTool.addClass("tool");
+
```

Comment:
you are repeating yourself a lot with the tool creation, it looks like the only difference is the img source. A function would have me this cleaner.